src/MagnumPlugins/TinyGltfImporter/TinyGltfImporter.cpp

```diff
@@ -0,0 +1,484 @@
+    const auto found = _d->nodesForName.find(name);
+    return found == _d->nodesForName.end() ? -1 : found->second;
+}
+
+std::unique_ptr<ObjectData3D> TinyGltfImporter::doObject3D(UnsignedInt id) {
+    const tinygltf::Node& node = _d->model.nodes[id];
+
+    std::vector<UnsignedInt> children(node.children.begin(), node.children.end());
+
+    Matrix4 transformation;
+
+    if(node.rotation.size() == 4) {
+        const auto vector = Vector3(node.rotation[0], node.rotation[1], node.rotation[2]);
+        const auto scalar = node.rotation[3];
+        transformation = Matrix4::from(Quaternion(vector, scalar).normalized().toMatrix(), {});
+    }
+
+    if(node.translation.size() == 3) {
+        const auto vector = Vector3(node.translation[0], node.translation[1], node.translation[2]);
+        Matrix4 translation = Matrix4::translation(vector);
+        transformation = transformation*translation;
+    }
+    if (node.scale.size() == 3) {
+        const auto vector = Vector3(node.scale[0], node.scale[1], node.scale[2]);
+        Matrix4 scale = Matrix4::scaling(vector);
+        transformation = transformation*scale;
+    }
+
+    /* node is a camera */
+    if(node.camera >= 0) {
+        UnsignedInt cameraId = node.camera;
+
+        return std::unique_ptr<ObjectData3D>{new ObjectData3D{children, transformation, ObjectInstanceType3D::Camera, cameraId, &node}};
+    /* node is a mesh */
+    } else if(node.mesh >= 0) {
+        UnsignedInt meshId = node.mesh;
+        /* TODO Multi-material models not supported */
+        Int materialId = _d->model.meshes[meshId].primitives[0].material;
+
+        return std::unique_ptr<ObjectData3D>{new MeshObjectData3D{children, transformation, meshId, materialId, &node}};
+    /* node is a light */
+    } else if(node.extLightsValues.find("light") != node.extLightsValues.end()) {
+        UnsignedInt lightId = node.extLightsValues.find("light")->second.number_array[0];
+
+        return std::unique_ptr<ObjectData3D>{new ObjectData3D{children, transformation, ObjectInstanceType3D::Light, lightId, &node}};
+    }
+
+    return std::unique_ptr<ObjectData3D>{new ObjectData3D{children, transformation, &node}};
+}
+
+UnsignedInt TinyGltfImporter::doMesh3DCount() const {
+    return _d->model.meshes.size();
+}
+
+Containers::Optional<MeshData3D> TinyGltfImporter::doMesh3D(const UnsignedInt id) {
+    const tinygltf::Mesh& mesh = _d->model.meshes[id];
+
+    /* primitive */
+    MeshPrimitive meshPrimitive = MeshPrimitive::Triangles;
+
+    if (mesh.primitives.size() > 1) {
+        Error() << "Trade::TinyGltfImporter::mesh3D(): currently only one primitive per mesh is supported";
+        return Containers::NullOpt;
+    }
+
+    const tinygltf::Primitive& primitive = mesh.primitives[0];
+
+    int mode = primitive.mode;
+    if (mode == TINYGLTF_MODE_POINTS) {
+        meshPrimitive = MeshPrimitive::Points;
+    } else if (mode == TINYGLTF_MODE_LINE) {
+        meshPrimitive = MeshPrimitive::Lines;
+    }
```

Comment:
What happens if `mode` is neither points, lines nor triangles? There's also triangle strip, triangle fan etc. I would personally do something like this:

```cpp
    MeshPrimitive meshPrimitive = MeshPrimitive{}; // (default-init to zero)
    if(primitive.mode == TINYGLTF_MODE_POINTS) {
        meshPrimitive = MeshPrimitive::Points;
    } else if(primitive.mode == TINYGLTF_MODE_LINE) {
        meshPrimitive = MeshPrimitive::Lines;
    } else if(primitive.mode == TINYGLTF_MODE_TRIANGLE) {
        meshPrimitive = MeshPrimitive::Triangles;
    ...
    } else {
        Error{} << "Trade::TinyGltfImporter::mesh3D(): unrecognized primitive" << primitive.mode;
        return Containers::NullOpt;
    }